src/debugAdapter/goDebug.ts

```diff
@@ -846,14 +846,29 @@ class GoDebugSession extends LoggingDebugSession {
 			return this.convertDebuggerPathToClient(pathToConvert);
 		}
 
-		// Fix for https://github.com/Microsoft/vscode-go/issues/1178
-		// When the pathToConvert is under GOROOT, replace the remote GOROOT with local GOROOT
+		// When the pathToConvert is under GOROOT or Go module cache, replace path appropriately
 		if (!pathToConvert.startsWith(this.delve.remotePath)) {
+			// Fix for https://github.com/Microsoft/vscode-go/issues/1178
 			const index = pathToConvert.indexOf(`${this.remotePathSeparator}src${this.remotePathSeparator}`);
 			const goroot = process.env['GOROOT'];
 			if (goroot && index > 0) {
 				return path.join(goroot, pathToConvert.substr(index));
 			}
+
+			const indexGoModCache = pathToConvert.indexOf(
+				`${this.remotePathSeparator}pkg${this.remotePathSeparator}mod${this.remotePathSeparator}`
+			);
+			const gopath = (process.env['GOPATH'] || '').split(path.delimiter)[0];
+
+			if (gopath && indexGoModCache > 0) {
+				return path.join(
+					gopath,
+					pathToConvert
+						.substr(indexGoModCache)
+						.split(this.remotePathSeparator)
+						.join(this.localPathSeparator)
```

Comment:
I just thought of one problem with this logic. Even if you are replacing the local path separator by the remote path separator, `path.join` will [normalize the path ](https://nodejs.org/api/path.html#path_path_join_paths) according to the platform. This means that you probably have to do the replacing part AFTER joining the path?

Reply:
We will have to do similar changes like you did to `toDebuggerPath` here as well. I was hoping to do that outside of this PR and have this PR focus only on the module cache part. Because the change has to applied to the goroot workaround and to the return statement a few lines below

Reply:
That sounds good.